### PR TITLE
Add focus indicator to File Details tabs view

### PR DESCRIPTION
**Pull request by kevgathuku**
This is a potential solution to #9944

It enables the outline by default so that the website is easily navigable via keyboard
Feedback appreciated 😄

### CODE REVIEW

**Comment by kevgathuku** [comment]
Some screenshots of the navigable UI after the fix:

Sharing tab:
screen shot 2018-06-21 at 21 54 53

File view:
screen shot 2018-06-21 at 21 54 28

**Comment by jancborchardt** [comment]
@kevgathuku I also thought about just restoring the outline, but it doesn’t look so nice. What we should do is simply also add the proper :focus/:active effect so it works the same as for :hover. :) Like for example we did with #9951 (Keyboard navigation accessibility)

**Comment by kevgathuku** [comment]
Thanks for the tips @jancborchardt 
I'll look into implementing a similar solution for this

**Comment by jancborchardt** [comment]
@kevgathuku great, thanks! :) Keep in mind to do it step by step, small separate pull requests. :) For example, start with fixing only the headers in the sidebar as you mentioned in the original issue.

**Comment by kevgathuku** [comment]
Thanks once again. I'll focus on that section for now @jancborchardt 

For the file details view the current hover effect is to underline the tab. I'm not sure the same underlining effect for the `focus` state would work very well. 
It might cause confusion between the focused and selected tabs.

screen shot 2018-06-22 at 00 01 53

For example, in this image, the `Comments` tab is the selected one / currently active.
The `Versions` tab is hovered / in focus. 

Restoring the browser default outline might just work, at least for the tabs section.

What do you think of this?  
In the meantime, I'm also exploring other ways to do it that might be less confusing.

**Comment by jancborchardt** [comment]
@kevgathuku I would say it’s two different steps: 1) visualizing :hover exactly as we do :focus and 2) thinking about how we want to visualize it.

I would say this pull request is about 1). :) And the current way to visualize is fine, but we can of course talk about it – but after we fix the basic accessibility.

**Comment by jancborchardt** [comment]
@kevgathuku in other words: What you show in the screenshot is great! :smiley:

**Comment by kevgathuku** [comment]
Got it @jancborchardt 
I'll focus on the first step with the current effect 👍

**Comment by kevgathuku** [comment]
Hi @jancborchardt would you happen to know if there is some guide on how to set up a local dev environment?, or a resource you can recommend? I'm using a Mac currently and I'm currently having trouble setting up the project for development.

I'm currently following the instructions at https://docs.nextcloud.com/server/12/admin_manual/installation/system_requirements.html#recommended-setup-for-running-nextcloud and https://docs.nextcloud.com/server/12/developer_manual/general/devenv.html
and I'm stuck with installing nextcloud on the apache server.
Also not sure if this is the right place to ask this kind of question? If it's not please, point me to the correct place 😄 
Thanks

**Comment by kevgathuku** [comment]
Works like a charm. Didn't know there was such a guide 😄 
Thanks @jancborchardt 👍

**Comment by jancborchardt** [comment]
> Works like a charm. Didn't know there was such a guide

Yeah, I need to add it to the docs :) good to know that it works for you too – I only checked it with friends macOSes cause I’m on Ubuntu. ;)

**Comment by kevgathuku** [comment]
This is once again ready for review 
Changes:
1. Enable focus indicator for file details tab. Current result with `Versions` tab focused:
screen shot 2018-06-25 at 10 28 43

2. Enable click indicator for the tabs i.e. when a tab is focused, clicking on `Spacebar` or `Enter` should select the tab
cc @jancborchardt

**Comment by jancborchardt** [comment]
@kevgathuku nice! :) cc @nextcloud/javascript for review too.

I just noticed that the underline is now grey whereas it used to be black, which is much more visible. We should change it back to that, using:
```
border-bottom: 1px solid var(--color-text-lighter);
```

Also, for better clickability the minimum size of the tab headers should be 44px, which can be achieved using:
```
.tabHeaders .tabHeader {
    padding: 12px;
}
```

Can you adjust that @kevgathuku? :) (Since you are member of the Nextcloud organization here on Github, you can also do branches in the server repository itself, then we can collaborate easily on branches. :) Don’t worry about breaking stuff, as the main branches are protected.)

File: core/css/apps.scss [hunk 1]
```diff
@@ -780,19 +780,12 @@ kbd {
 			font-weight: 600;
 			border-bottom: 1px solid var(--color-border);
```
Comment (jancborchardt) [inline core/css/apps.scss]: This line can be removed because the border-bottom for .selected is set in line 788 down below. :)

**Comment by kevgathuku** [comment]
Thanks for the pointers @jancborchardt 
In future I'll create all branches on the repo.  I agree, it will make collaborating much easier 😄 

Finally, I've made the requested changes

**Comment by MorrisJobke** [comment]
cc @nextcloud/designers

**Review by jancborchardt — APPROVED** [review]
This is super awesome @kevgathuku! :) Just wrote a comment on a file using keyboard only and it worked well. :tada: 

cc @nextcloud/accessibility for review too :)

**Review by skjnldsv — APPROVED** [review]

**Comment by kevgathuku** [comment]
Finally got merged :tada:  Thanks for all the feedback @jancborchardt 
Happy to contribute more :smile:

**Comment by skjnldsv** [comment]
> Finally got merged

5 days is pretty quick for us :D

**Comment by jancborchardt** [comment]
@kevgathuku awesome work! :) It would also be great to have you at our Nextcloud Conference in Berlin late August: https://nextcloud.com/conf – we provide up to 80% travel/accomodation support for contributors! :tada:

**Comment by kevgathuku** [comment]
Amazing! I'll check out the conference and apply.

**Pull request merged**